deployer.py

```diff
@@ -17,103 +41,41 @@ class Deployer(BaseModel):
-        )
-
-    def build_nginx(self):
-        """Build the proxy nginx docker image"""
-        self.run(f'docker build ov-nginx')
-
-    def push_ov_wag(self):
-        """Push the backend ov_wag docker image to hub.docker.com
+        run(f'kubectl config use-context {self.context}')
 
-        Requires the user to be logged in"""
-        self.run(f'docker push {self.ov_wag_tag}')
+    def _deploy(self, repo_name, tag, src=''):
+        """Deploy helper function
 
-    def push_ov_frontend(self):
-        """Push the frontend ov-frontend docker image to hub.docker.com
-
-        Requires the user to be logged in"""
-        self.run(f'docker push {self.ov_frontend_tag}')
-
-    def push_nginx(self):
-        """Push the proxy nginx docker image to hub.docker.com
-
-        Requires the user to be logged in"""
-        self.run(f'docker push {self.ov_nginx_tag}')
-
-    def update_ov_wag_workload(self):
-        """Sets the backend pod image to"""
-        self.run(f'kubectl set image deployment.apps/ov ov={self.ov_wag_tag}')
-
-    def update_ov_frontend_workload(self):
-        self.run(
-            f'kubectl set image deployment.apps/ov-frontend ov-frontend={self.ov_frontend_tag}'
-        )
-
-    def deploy_ov_wag(self):
-        """Deploy the backend"""
-        print(f'Deploying ov-wag: "{self.ov_wag}"')
-        self.build_ov_wag()
-        self.push_ov_wag()
-        self.update_ov_wag_workload()
-
-    def deploy_ov_frontend(self):
-        """Deploy the frontend"""
-        print(f'Deploying ov-frontend: "{self.ov_frontend}"')
-        self.build_ov_frontend()
-        self.push_ov_frontend()
-        self.update_ov_frontend_workload()
-
-    def deploy_ov_nginx(self):
-        """Deploy the proxy
-
-        TODO: deploy proxy"""
-        print(f'Deploying ov_nginx: "{self.ov_nginx}"')
+        Deploy an individual image from a repo and tag name"""
+        build_image(repo_name, tag, src=src)
+        push_image(repo_name, tag)
+        update_workload(pod=repo_name, tag=tag)
 
     def deploy(self):
-        """Run the deployment process using the current context"""
+        """Deploy all
+
+        Run the full deployer process using the current context"""
         print(f'Starting deployment using context "{self.context}"')
 
         if not any([self.ov_wag, self.ov_frontend, self.ov_nginx]):
             raise Exception(f'Nothing specified for deployment.')
         if self.ov_wag:
-            self.deploy_ov_wag()
+            self._deploy('ov-wag', self.ov_wag, src=f'{OV_WAG_URL}#{self.ov_wag}')
         if self.ov_frontend:
-            self.deploy_ov_frontend()
+            self._deploy(
+                'ov-frontend',
+                self.ov_frontend,
+                src=f'{OV_FRONT_URL}#{self.ov_frontend}',
+            )
         if self.ov_nginx:
-            self.deploy_ov_nginx()
+            self._deploy('ov-nginx', self.ov_nginx, src='ov-nginx')
```

Comment:
Unnecessary here.

Reply:
If we were to split `ov-nginx` into it's own repo, we would need to explicitly point the `src` to the github repo.

When we add the `jumpbox` options to the CLI, it will not need the `src` since it's building from a local file... until we split it into it's own repo.